raving-remoras/bangazon-site
# Product List and Product Details

# Description
This PR covers functionality related to viewing all products and the specific details page of an individual product.

## Related Ticket(s)
Ticket #14 : User can view product detail

## Proposed Changes
Changes to the structure of the views files. All authorization related views have been broken into their own file, index views are in their own file and product based views are in their own file. 

Changes to the structure of the templates. Moved product list and product detail templates into the main templates folder.

## Steps to Test Solution

1. Once logged in, when you select the shop item in the navbar, you should be taken to a view of all products with a button to view details.
1. If you select view details, you should be taken to a page with the product title, description, the available quantity (the original quantity minus the completed orders with that product ID), the price and an add to cart button.
1. If the quantity is 0, the add to cart button should be deactivated, and a badge showing that the product is sold out
1. If the quantity is less than 5, there should be a badge displaying that the product is almost sold out
1. If anyone has the product in their cart (an incomplete order) it will display a message below the add to cart button.

## Testing


## Documentation
